### DISCUSSION TITLE
Extract a number from a piece of text - more training data, perform worse

### DISCUSSION THREAD

**Discussion by cwei-bgl** [google-research/text-to-text-transfer-transformer#949] [Q&A]
Hi,

I have a simple task to extract a number from a piece of text. For example, "I bought 2.5 kg apples for 6.8 dollars today from the stop." and I would like to extract '2.5'. I sampled a 2000 training data and I could train a T5 small model to get a good accuracy score 0.97. But when I used the whole data set  about 100,000 and accuracy is around 0.21(and stuck). I tried to use the model trained from 2000 training data and did the prediction on the whole data set. The performance is prefer good and close to expected accuracy 0.97. I wonder what the problem is with the whole data set?

I am thinking if sequential labeling(token classification) is more suitable in this case. 

-----
I tried again this morning with 10,000 training data. It also performed good. 

`
'train-loss': 0.0268, 'eval-loss': 0.027098597958683968, 'eval-rougeL-precision': 0.98775, 'eval-rougeL-recall': 0.99025, 'eval-rougeL-f1': 0.9886666666666668, 'eval-accuracy': 0.984, 'eval-support': 2000, "epoch": 10.0
`
The accuracy was calculated by comparing the predicted label text with the true label text. Only full match counts as correct. `0.984` means 2 out of 100 was incorrect, which is pretty good result.

However, with full data set of about 100,000 data points, the recall and accuracy was stuck.
`
'train-loss': 0.0002, "eval-loss": 4.71308158012107e-05, "eval-rougeL-precision": 0.9675825284754389, "eval-rougeL-recall": 0.5387609833306606, "eval-rougeL-f1": 0.6517330991444945, "eval-accuracy": 0.22051639916259597, "eval-support": 24361, "epoch": 10.0
`

I think it(the reason for stuck) might be to do with the train-loss is too small in full data set case('train-loss': 0.0002). I can see in the source code that it is using `CrossEntropyLoss`. What I don't understand is why with full data set, the train-loss became small? CrossEntropyLoss compare the difference between two distributions. I expect average losses to be similar between partial data set and full data set.

`
loss_fct = CrossEntropyLoss(ignore_index=-100)
loss = loss_fct(lm_logits.view(-1, lm_logits.size(-1)), labels.view(-1))
`

Any suggestion is highly appreciated. 

Thanks, Cheng
